pi-apps-terminal-bash-edition.sh

```diff
@@ -148,106 +90,154 @@ while [ "$1" != "" ]; do
-			ls $PI_APPS_DIR | grep --color=always -vx "$(ls "${PI_APPS_DIR}/data/status" | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
+			list_apps uninstalled
 			exit $?
 		;;
 		list-corrupted)
 			#list all the corrupted apps
-			#list_apps corrupted
-			ls $PI_APPS_DIR/apps | grep --color=always -x "$(grep -rx 'corrupted' "${PI_APPS_DIR}/data/status" | awk -F: '{print $1}' | sed 's!.*/!!' | sed -z 's/\n/\\|/g' | sed -z 's/\\|$/\n/g')"
+			list_apps corrupted
 			exit $?
 		;;
 		list-all)
 			#list all the apps
-			list-all
+			list_apps cpu_installable
 			exit $?
 		;;
 		search)
 			shift
-			args="$*"
-			#search apps
-			search "$args"
+			echo
+			while read -r line; do
+				[[ -z "$line" ]] && continue
+				echo -e "${bold}${inverted}${light_blue}${underline}$line${normal}\n"
+				echo -e "${green}$(cat "$DIRECTORY/apps/$line/description" || echo "No description available")${normal}\n"
+			done < <(app_search "$1" 2>/dev/null)
 			exit $?
 		;;
-		update-apps)
-			#update all pi-apps apps
-			$PI_APPS_DIR/manage update-all
-			exit $?
-			;;
 		update)
-			#update all pi-apps
-			$PI_APPS_DIR/updater cli
+			#update pi-apps
+			shift
+			if [ "$*" == "cli" ] || [ "$*" == "gui" ] || [ "$*" == "cli-yes" ] || [ "$*" == "gui-yes" ] || [ "$*" == "" ]; then
+				"$DIRECTORY/updater" "$*" 
+			else
+				error "Argument '$*' not available. \n    Accepted arguments: [gui|cli|gui-yes|cli-yes]"
+			fi
 			exit $?
 		;;
-		website)
+		info)
 			shift
-			[[ "$@" == "" ]] && error "'website' option passed, but no app provided!"
-			args="$*"
-			#print the website of a app
-			get-website "$args" 2>/dev/null
-			if [[ "$website_error" == "1" ]]; then
-				echo -e "${red}${bold}ERROR:${normal}${red} There is no app called ${light_red}'$@'${red}!${normal}"
-				exit 1
+			
+			if [ -z "$*" ]; then
+				error "No app provided."
 			else
-				echo -e "${cyan}${inverted}$@'s website:${normal}"
-				echo -e "${bold}$website${normal}"
-				exit 0
+				app_name="$*"
+			fi
+				
+			if ! [ -d "$DIRECTORY/apps/$app_name" ]; then
+				error "App not found."
+			fi
+			
+			description="$(cat "${DIRECTORY}/apps/$app_name/description" || echo 'Description unavailable')"
+
+			abovetext="
+- Current status: $(app_status "$app_name") | sed 's/corrupted/corrupted (installation failed)/g' | sed 's/disabled/disabled (installation is prevented on your system)/g')"
+			
+			if [ -f "${DIRECTORY}/apps/$app_name/website" ];then
+				#show website if it exists
+				abovetext="$abovetext
+- Website: $(head -n1 < "${DIRECTORY}/apps/$app_name/website")"
+
+			fi
+
+			if [ -z "$clicklist" ];then
+				clicklist="$(usercount)"
+			fi
+
+			usercount="$(echo "$clicklist" | grep " $app_name"'$' | awk '{print $1}' | head -n1)"
+			if [ -n "$usercount" ] && [ "$usercount" -gt 20 ];then
+				abovetext="$abovetext
+- $(printf "%'d" "$usercount") users"
```

Comment:
what does the `'` in between the `%` and `d` in the `printf` call do?

Reply:
I don't know. I copied them from [the pi-apps `gui` script](https://github.com/Botspot/pi-apps/blob/7318c9e4285167970c2d26e2159fcf1b5b2c9fc3/gui#L525-L551).

Reply:
It doesn't seem to do anything.
Coild you remove it please?

Reply:
Yes, it does something:
`%d, %i - Print the argument as a signed decimal integer.`
It will convert `+123` to `123`.

Since it doesn't affect the output, I don't see any reason to remove it.

Reply:
I meant the `'`, it doesn't affect the output, but doesn't do anything as far as I know.
so could you remove it please?

Reply:
Removed.